From stable-bounces@linux.kernel.org Sun Apr 23 16:21:19 2006
Message-Id: <200604232320.k3NNKs3V001090@shell0.pdx.osdl.net>
To: a1426z@gawab.com, alan@lxorguk.ukuu.org.uk,
        B.Zolnierkiewicz@elka.pw.edu.pl, stable@kernel.org,
        mm-commits@vger.kernel.org
From: akpm@osdl.org
Date: Sun, 23 Apr 2006 16:19:45 -0700
Cc: 
Subject: ide-io: increase timeout value to allow for slave wakeup

From: Al Boldi <a1426z@gawab.com>

During an STR resume cycle, the ide master disk times-out when there is
also a slave present (especially CD).  Increasing the timeout in ide-io
from 10,000 to 100,000 fixes this problem.

Cc: Alan Cox <alan@lxorguk.ukuu.org.uk>
Cc: Bartlomiej Zolnierkiewicz <B.Zolnierkiewicz@elka.pw.edu.pl>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/ide/ide-io.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- linux-2.6.16.11.orig/drivers/ide/ide-io.c
+++ linux-2.6.16.11/drivers/ide/ide-io.c
@@ -932,7 +932,7 @@ static ide_startstop_t start_request (id
 			printk(KERN_WARNING "%s: bus not ready on wakeup\n", drive->name);
 		SELECT_DRIVE(drive);
 		HWIF(drive)->OUTB(8, HWIF(drive)->io_ports[IDE_CONTROL_OFFSET]);
-		rc = ide_wait_not_busy(HWIF(drive), 10000);
+		rc = ide_wait_not_busy(HWIF(drive), 100000);
 		if (rc)
 			printk(KERN_WARNING "%s: drive not ready on wakeup\n", drive->name);
 	}
